lib/api_auth/base.rb

```diff
@@ -22,22 +22,24 @@ class << self
     # access_id: The public unique identifier for the client
     #
     # secret_key: assigned secret key that is known to both parties
-    def sign!(request, access_id, secret_key)
+    def sign!(request, access_id, secret_key, options = {})
+      options = { :override_http_method => nil, :with_http_method => false }.merge(options)
```

Comment:
The merge of `:override_http_method` is confusing here. There's no need to set it to `nil` since if it's not there it'll be seen as `nil` anyway. Arguably you don't need to do this for `:with_http_method` either because you're not comparing it directly to a boolean.